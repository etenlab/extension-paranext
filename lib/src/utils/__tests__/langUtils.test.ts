import { LanguageInfo } from '@eten-lab/ui-kit';
import {
  compareLangInfo,
  langInfo2tag,
  subTags2LangInfo,
  tag2langInfo,
} from '../langUtils';

describe('langUtils tests', () => {
  describe('tag2langInfo', () => {
    it('convetrs language tag to LanguageInfo object', () => {
      const tag = 'en-UA-basiceng';
      const res = tag2langInfo(tag);
      expect(res).toMatchSnapshot();
    });
  });
  describe('langInfo2tag', () => {
    it('convetrs LanguageInfo object to tag', () => {
      const langInfo: LanguageInfo = {
        dialect: {
          descriptions: ['doesn`t matter'],
          tag: 'basiceng',
        },
        lang: {
          descriptions: ['doesn`t matter'],
          tag: 'en',
        },
        region: {
          descriptions: ['doesn`t matter'],
          tag: 'UA',
        },
      };
      const res = langInfo2tag(langInfo);
      expect(res).toMatchSnapshot();
    });
  });
  describe('subTags2LangInfo', () => {
    it('convetrs subTags to LangInfo', () => {
      const langSubtag = 'en';
      const regionSubtag = 'UA';
      const dialectSubtag = 'basiceng';
      const res = subTags2LangInfo({
        lang: langSubtag,
        region: regionSubtag,
        dialect: dialectSubtag,
      });
      expect(res).toMatchSnapshot();
    });
  });
  describe('compareLangInfo', () => {
    it('compares two LangInfo objects', () => {
      const langSubtag = 'en';
      const regionSubtag = 'UA';
      const dialectSubtag = 'basiceng';
      const langInfo1 = subTags2LangInfo({
        lang: langSubtag,
        region: regionSubtag,
        dialect: dialectSubtag,
      });
      const langInfo2 = subTags2LangInfo({
        lang: langSubtag,
        region: regionSubtag,
        dialect: dialectSubtag,
      });
      expect(compareLangInfo(langInfo1, langInfo2)).toBe(true);
    });
  });
});
