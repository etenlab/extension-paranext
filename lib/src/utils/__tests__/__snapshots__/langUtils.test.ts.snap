// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`langUtils tests langInfo2tag convetrs LanguageInfo object to tag 1`] = `"en-UA-basiceng"`;

exports[`langUtils tests subTags2LangInfo convetrs subTags to LangInfo 1`] = `
Object {
  "dialect": Object {
    "descriptions": Array [
      "Basic English",
    ],
    "tag": "basiceng",
  },
  "lang": Object {
    "descriptions": Array [
      "English",
    ],
    "tag": "en",
  },
  "region": Object {
    "descriptions": Array [
      "Ukraine",
    ],
    "tag": "UA",
  },
}
`;

exports[`langUtils tests tag2langInfo convetrs language tag to LanguageInfo object 1`] = `
Object {
  "dialect": Object {
    "descriptions": Array [
      "Basic English",
    ],
    "tag": "basiceng",
  },
  "lang": Object {
    "descriptions": Array [
      "English",
    ],
    "tag": "en",
  },
  "region": Object {
    "descriptions": Array [
      "Ukraine",
    ],
    "tag": "UA",
  },
}
`;
